File: benchmarks/array_sum.swift
// Dynamic array push + sum — measures collection/runtime overhead

let n: Int64 = 10_000_000
var arr: [Int64] = []

for i: Int64 in 0..<n {
    arr.append(i)
}

var sum: Int64 = 0
for x in arr {
    sum += x
}

print(sum)

File: benchmarks/array_sum.c
// Dynamic array push + sum — measures collection/runtime overhead
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>
#include <inttypes.h>

int main(void) {
    int64_t n = 10000000;
    int64_t cap = 16;
    int64_t len = 0;
    int64_t *arr = malloc(cap * sizeof(int64_t));

    for (int64_t i = 0; i < n; i++) {
        if (len >= cap) {
            cap *= 2;
            arr = realloc(arr, cap * sizeof(int64_t));
        }
        arr[len++] = i;
    }

    int64_t sum = 0;
    for (int64_t i = 0; i < len; i++) {
        sum += arr[i];
    }

    printf("%" PRId64 "\n", sum);
    free(arr);
    return 0;
}
